kitsune/wiki/managers.py

```diff
@@ -12,26 +12,77 @@ class VisibilityManager(models.Manager):
     # should be "document".
     document_relation: str | None = None
 
+    @property
+    def document_relation_prefix(self):
+        """Provides ORM prefix for the document relation."""
+        return f"{self.document_relation}__" if self.document_relation else ""
+
     def get_creator_condition(self, user):
         """
         Return a conditional (e.g., a Q or Exists object) that is only true
         when the given user is the creator of this document or revision.
         """
         raise NotImplementedError
 
-    def visible(self, user, **kwargs):
+    def unrestricted(self, user=None, **kwargs):
+        """
+        Documents are unrestricted for a given user if the user is a superuser, or
+        if they're not restricted to staff or a specific group, or when the user is
+        staff and they're restricted by staff, or when the user is a member of the
+        specific group to which they're restricted. A translation (i.e., a document
+        with a parent), follows its parent's restrictions.
+        """
+        if user and user.is_superuser:
+            return self.filter(**kwargs)
+
+        prefix = self.document_relation_prefix
+
+        def unrestricted_condition(parent_prefix=""):
+            """
+            Create the unrestricted condition for either the document or
+            the document's parent.
+            """
+            restrict_to_staff = f"{prefix}{parent_prefix}restrict_to_staff"
+            restrict_to_group = f"{prefix}{parent_prefix}restrict_to_group"
+
+            parent_condition = Q(**{f"{prefix}parent__isnull": not bool(parent_prefix)})
+            not_restricted = Q(**{restrict_to_staff: False}) & Q(
```

Comment:
Why aren't you using `filter/exclude(restrict_to_staff=False, restrict_to_group__isnull=True)?

Reply:
It's because this inner function has to return a Q object, because ultimately, in the outer method, the Q object it returns is used within the "or" condition `unrestricted_condition() | unrestricted_condition("parent__")`.